kimchi/wasm/src/rayon.rs

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2022 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use js_sys::JsString;
+use js_sys::Promise;
+use spmc::{channel, Receiver, Sender};
+use wasm_bindgen::prelude::*;
+
+static mut THREAD_POOL: Option<rayon::ThreadPool> = None;
```

Comment:
this is the new `rayon` lib! at its heart is a mutable global `THREAD_POOL` which holds an option (our current thread pool, or nothing). when we set it to `None`, Rust cleans up the thread pool / workers for us